src/bots/depositor.py

```diff
@@ -274,3 +275,57 @@ def _send_deposit_tx(
             self._flashbots_works = True
 
         return success
+
+    def _prepare_transaction(
+        self,
+        block_number: int,
+        block_hash: Hash32,
+        deposit_root: Hash32,
+        staking_module_id: int,
+        staking_module_nonce: int,
+        payload: bytes,
+        guardian_signs: tuple[tuple[str, str], ...]
+    ) -> ContractFunction:
+        """
+        It either follows a regular flow or builds a direct deposit transaction.
+
+        Conditions to build direct deposit transaction are:
+        1. Env variable is set
+        2. balance in the vault is not 0
+        3. The calls responded without errors
+        """
+        deposit_tx = self.w3.lido.deposit_security_module.deposit_buffered_ether(
+            block_number,
+            block_hash,
+            deposit_root,
+            staking_module_id,
+            staking_module_nonce,
+            payload,
+            guardian_signs,
+        )
+        if variables.is_mellow_strategy_set():
+            try:
+                vault_address = self.w3.lido.simple_dvt_staking_strategy.vault()
```

Comment:
Lets move everything related to Mellow to separate function.
```python
try:
    if _self.is_mellow_depositable(staking_module_id):
        deposit_tx = self._prepare_mellow_deposit_tx(...)
except:
    log(exception)
else:
    log('Mellow tx')

```

Reply:
I have done something similar, but I need to reassign `deposit_tx`

```
func mellow_transaction(pass_all_the_parameters) -> Option[CallableFunction]:
    ...

func build_transacetion(all_the_parameters) -> CallableFunction:
   o = mellow_transaction(all_the_parameters)
   if o is not None:
       return o
   else:
       return deposit(all_the_parameters)
```

seems more difficult